### PR TITLE
Fix looking behavior

### PR DESCRIPTION
**Pull request by ph0t0shop**
See https://github.com/PrismarineJS/mineflayer-pathfinder/issues/81

### CODE REVIEW

File: lib/plugins/physics.js [hunk 1]
```diff
@@ -196,26 +195,37 @@ function inject (bot) {
...
     }
 
-    do {
-      await once(bot, 'move')
-    } while (Math.abs(deltaYaw(bot.entity.yaw, lastSentYaw)) >= 0.001)
+    await lookingTask.promise
   })
 
-  bot.lookAt = callbackify(async (point, force) => {
+  bot.lookAt = (point, force, cb) => {
```
Comment (rom1504) [inline lib/plugins/physics.js:222]: why did you add back the cb ?

File: lib/plugins/physics.js [hunk 2]
```diff
@@ -196,26 +195,37 @@ function inject (bot) {
     })
   }
 
+  let lookingTask = createDoneTask()
+
+  bot.on('move', () => {
+    if (Math.abs(deltaYaw(bot.entity.yaw, lastSentYaw)) < 0.001 && !lookingTask.done) {
```
Comment (rom1504) [inline lib/plugins/physics.js:201]: better put the lookingTask.done first in the condition so it doesn't do the computation if not needed

**Comment by rom1504** [comment]
left 2 comments, but overall it looks good, I like the use of the task here

File: lib/plugins/physics.js (hunk 1 above)
Comment (ph0t0shop) [reply lib/plugins/physics.js:222]: Because this is just the lookAt function, it is a proxy function for the look function. All the functions still take cb, but this one I made it just return the promise of look() after calculating those values (yaw and pitch). Otherwise it needed an await in there.

File: lib/plugins/physics.js (hunk 2 above)
Comment (ph0t0shop) [reply lib/plugins/physics.js]: You're right... short-circuiting for the win

File: lib/plugins/physics.js (hunk 1 above)
Comment (rom1504) [reply lib/plugins/physics.js:222]: what's the problem for having an await ?

Comment (ph0t0shop) [reply lib/plugins/physics.js:222]: Not really a problem, this just seemed the most pragmatic, it being a proxy function it doesn't really need to be async.

**Comment by rom1504** [comment]
@ph0t0shop to clarify, https://gist.github.com/rom1504/6a9127764b43e38569e704f94ac99908 will always print a b c

**Comment by ph0t0shop** [comment]
@rom1504 yeah I understand that now, it just seems unnecessary wrapping a promise inside another promise. But we can do it if you want

File: lib/plugins/physics.js (hunk 1 above)
Comment (rom1504) [reply lib/plugins/physics.js:222]: the problem with doing it that way is that it will make it harder to remove callbacks the day we want to
I think it's better in mineflayer internals to never call functions with their callbacks
and to callbackify them from the outside
That way removing the callback can be done simply by removing the callbackify, nothing else

Comment (ph0t0shop) [reply lib/plugins/physics.js]: Ok fine, done

**Review by rom1504 — APPROVED** [review]
